### PR TITLE
Added the ability to paste in search mode

### LINKED ISSUE #7511: Paste in Vi Search with key binding
I haven't seen a key binding that allows pasting in vi search, like `Control+Shift+V` in normal search. That could be useful.

### PR DESCRIPTION
**Pull request by omagdy7**
Fixes #7511

### CODE REVIEW

**Review by kchibisov — CHANGES_REQUESTED** [review]
This is not right way to solve this, since now you paste `twice` into normal search prompt.

Probably the solution would be for each paste binding and a paste binding in `vi | search` mode.

**Comment by omagdy7** [comment]
> This is not right way to solve this, since now you paste `twice` into normal search prompt.

When I tested the change only one paste happens when I press "CTRL+SHIFT+V", Or are you referring to something else?

demo

**Comment by omagdy7** [comment]
Ok I think I got it normal search mode without vi gets triggered twice oh my bad

**Review by chrisduerr — CHANGES_REQUESTED** [review]
Missing a changelog entry.

File: alacritty/src/config/bindings.rs [hunk 1]
```diff
@@ -546,7 +546,7 @@ fn common_keybindings() -> Vec<KeyBinding> {
     bindings!(
         KeyBinding;
         "v",    ModifiersState::CONTROL | ModifiersState::SHIFT, ~BindingMode::VI;                       Action::Paste;
-        "v",    ModifiersState::CONTROL | ModifiersState::SHIFT, +BindingMode::SEARCH;                   Action::Paste;
+        "v",    ModifiersState::CONTROL | ModifiersState::SHIFT, +BindingMode::VI, +BindingMode::SEARCH; Action::Paste;
```
Comment (kchibisov) [inline alacritty/src/config/bindings.rs:549]: You should do the same for all other bindings `Paste` bindings, as well as document them in the `alacritty-binding.5.scd` man page.

**Review by kchibisov — CHANGES_REQUESTED** [review]

**Comment by omagdy7** [comment]
The alacritty-bindings man syntax was really confusing I hope I got it right but what kind of format is that? I've never seen it before

File: CHANGELOG.md [hunk 2]
```diff
@@ -22,6 +22,10 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 - IME input lagging behind on X11
 - xdotool modifiers input not working correctly on X11
 
+### Added
+
+- Support for pasting in Vi + Search mode
```
Comment (chrisduerr) [inline CHANGELOG.md:27]: Please check the correct section order at the top of the file.

**Review by chrisduerr — CHANGES_REQUESTED** [review]

**Review by chrisduerr — CHANGES_REQUESTED** [review]

File: extra/man/alacritty-bindings.5.scd [hunk 3]
```diff
@@ -442,6 +450,10 @@ configuration. See *alacritty*(5) for full configuration format documentation.
 :  _"Command"_
 :[
 :  _"Paste"_
```
Comment (chrisduerr) [inline extra/man/alacritty-bindings.5.scd:452]: This old binding doesn't actually show the mode as `~Vi`, could you fix that?

**Review by chrisduerr — CHANGES_REQUESTED** [review]

**Review by chrisduerr — APPROVED** [review]
Thank you for the PR.

**Pull request merged**